portalapi/controllers/routes.py

```diff
@@ -313,6 +313,13 @@ def interp(at, left, right):
   left = fetch(lrm["left"], 1)
   right =  fetch(lrm["right"], 1)
   middle = fetch(lrm["middle"])
+
+  if len(left)<1:
```

Comment:
I'm not sure I have a preference. This just seemed to mirror the check for middle, which could have also been done differently.